src/main/java/com/spotify/docker/client/DockerHost.java

```diff
@@ -17,52 +17,73 @@
 
 package com.spotify.docker.client;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.net.HostAndPort;
 
 import java.net.URI;
+import java.nio.file.Paths;
 import java.util.Locale;
 
-import static com.google.common.base.Optional.fromNullable;
+import static com.google.common.base.MoreObjects.firstNonNull;
 import static com.google.common.base.Strings.isNullOrEmpty;
-import static java.lang.System.getenv;
 
 /**
  * Represents a dockerd endpoint. A codified DOCKER_HOST.
  */
 public class DockerHost {
 
-  public static final int DEFAULT_PORT = 2375;
-  public static final String DEFAULT_HOST = "localhost";
-  public static final String DEFAULT_UNIX_ENDPOINT = "unix:///var/run/docker.sock";
+  /**
+   * An interface to be mocked during testing.
+   */
+  @VisibleForTesting
+  interface SystemDelegate {
+    String getProperty(String key);
+    String getenv(String name);
+  }
+
+  private static SystemDelegate systemDelegate = new SystemDelegate() {
```

Comment:
Having this as a static attribute that you can set during tests seems bad, but if this is an instance attribute, all the currently static methods like [`endpointFromEnv()`](https://github.com/spotify/docker-client/pull/441/files#diff-0d0cb499d121de8c7a4e50c2a2459ec9R156) would have to go into another static class in order to be used by the static factory method `fromEnv()`. Since `setSystemDelegate()` is package local, I thought this is OK although inelegant.